src/version.rs

```diff
@@ -58,6 +61,39 @@ impl Default for HttpVersion {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<http_types::Version> for HttpVersion {
+    fn from(v: http_types::Version) -> HttpVersion {
+        match v {
+            http_types::Version::HTTP_09 =>
+                HttpVersion::Http09,
+            http_types::Version::HTTP_10 =>
+                HttpVersion::Http10,
+            http_types::Version::HTTP_11 =>
+                HttpVersion::Http11,
+            http_types::Version::HTTP_2 =>
+                HttpVersion::H2
+        }
+    }
+}
+
+#[cfg(feature = "compat")]
+impl From<HttpVersion> for http_types::Version {
+    fn from(v: HttpVersion) -> http_types::Version {
+        match v {
+            HttpVersion::Http09 =>
+                http_types::Version::HTTP_09,
+            HttpVersion::Http10 =>
+                http_types::Version::HTTP_10,
+            HttpVersion::Http11 =>
+                http_types::Version::HTTP_11,
+            HttpVersion::H2 =>
+                http_types::Version::HTTP_2,
+            _ => panic!("attempted to convert unexpected http version")
```

Comment:
The `HttpVersion::__DontMatchMe` variant can be used here, so we know it shouldn't ever exist, and then use `unreachable!` instead of `panic!`.

Reply:
The problem is that hyper has made `H2` and `H2c` two distinct values of `HttpVersion`, rather than conveying non-cleartext/cleartext via some other channel. `http` on the other hand does not make the distinction, which is why this match is not exhaustive and the function therefore partial.

Reply:
That's a good point, I had forgotten about that variant. Woops.